lib/active_model_serializers/model_mixin.rb

```diff
@@ -0,0 +1,57 @@
+# ActiveModelSerializers::Model is a convenient
+# serializable class to inherit from when making
+# serializable non-activerecord objects.
+module ActiveModelSerializers
+  module ModelMixin
+    def self.included(klass)
+      klass.class_eval do
+        include ActiveModel::Model
+        include ActiveModel::Serializers::JSON
+
+        def read_attribute_for_serialization(key)
+          if key == :id || key == 'id'
+            send(key)
+          else
+            if is_a?(ActiveModelSerializers::Model)
+              # Support legacy behavior
+              attributes[key] || (send(key) if respond_to?(key))
+            else
+              send(key) if respond_to?(key)
+            end
+          end
+        end
+      end
+    end
+
+    #def id
+      #self.class.name.downcase
+    #end
+
+    #def cache_key
+    #end
+
+    #def errors
+      #@errors ||= ActiveModel::Errors.new(self)
+    #end
```

Comment:
obviously this is a wip, but can't must remove these. They're part of the [contract](https://github.com/rails-api/active_model_serializers/blob/v0.10.2/lib/active_model/serializer/lint.rb)